src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs

```diff
@@ -136,23 +136,29 @@ private static TMetrics GetAverageMetrics(IEnumerable<TMetrics> metrics, TMetric
                 Contracts.Assert(newMetrics != null);
 
                 var result = new RegressionMetrics(
-                    l1: GetAverageOfNonNaNScores(newMetrics.Select(x => x.MeanAbsoluteError)),
-                    l2: GetAverageOfNonNaNScores(newMetrics.Select(x => x.MeanSquaredError)),
-                    rms: GetAverageOfNonNaNScores(newMetrics.Select(x => x.RootMeanSquaredError)),
-                    lossFunction: GetAverageOfNonNaNScores(newMetrics.Select(x => x.LossFunction)),
-                    rSquared: GetAverageOfNonNaNScores(newMetrics.Select(x => x.RSquared)));
+                    l1: GetAverageOfNonNaNAndNonInfinityScores(newMetrics.Select(x => x.MeanAbsoluteError)),
+                    l2: GetAverageOfNonNaNAndNonInfinityScores(newMetrics.Select(x => x.MeanSquaredError)),
+                    rms: GetAverageOfNonNaNAndNonInfinityScores(newMetrics.Select(x => x.RootMeanSquaredError)),
+                    lossFunction: GetAverageOfNonNaNAndNonInfinityScores(newMetrics.Select(x => x.LossFunction)),
+                    rSquared: GetAverageOfNonNaNAndNonInfinityScores(newMetrics.Select(x => x.RSquared)));
                 return result as TMetrics;
             }
 
             throw new NotImplementedException($"Metric {typeof(TMetrics)} not implemented");
         }
 
-        private static double GetAverageOfNonNaNScores(IEnumerable<double> results)
+        /// <summary>
+        /// return the average of non-nan and non-infinity scores from <paramref name="results"/>, if all scores are nan/infinity, nan will be returned.
+        /// </summary>
+        /// <param name="results"></param>
+        /// <returns></returns>
+        private static double GetAverageOfNonNaNAndNonInfinityScores(IEnumerable<double> results)
         {
-            var newResults = results.Where(r => !double.IsNaN(r));
-            // Return NaN iff all scores are NaN
+            var newResults = results.Where(r => !double.IsNaN(r) && !double.IsInfinity(r));
+            // Return NaN iff all scores are NaN or Infinity
             if (newResults.Count() == 0)
                 return double.NaN;
+
             // Return average of non-NaN scores otherwise
```

Comment:
What are the cases being fixed by removing the +/-Infinity values? I'm not sure they should be filtered, as +/-Infinity can be a correct, and wanted, metric value. For instance, for log-loss reduction. 

Instead of filtering you could modify [GetIndexClosestToAverage()](https://github.com/dotnet/machinelearning/blob/f87a3bbd8adb12934dae0a0060813ce9b7500664/src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs#L160-L182) to properly handle the average being +/-Infinity. 

Perhaps coded as:
```c#
 private static int GetIndexClosestToAverage(IEnumerable<double> values, double average) 
 { 
     // Average will be NaN iff all values are NaN. 
     // Return the first index in this case. 
     if (double.IsNaN(average)) 
         return 0; 

    /* -- New checks -- */

    // Return the largest metric value when the average is +Infinity
    if (double.IsPositiveInfinity(average))
        return values.IndexOf(values.Max());

    // Return the smallest metric value when the average is -Infinity
    if (double.IsNegativeInfinity(average))
        return values.IndexOf(values.Min());

   /* ...function continues as before... */
```

Reply:
It's the same reason why we filter nan value out before calculating average score, there is no mathematical meaning to include infinite value either, thinking about what's the average score of [+inf, -1, 0, 1, -inf] should be.

If we want to keep +/- inf when calculating average score, we'd better keep nan as well. In that way, the evaluation for cross-validation runner is closer to ground truth.

Reply:
I think the right answer is to keep the +/-Infinity. 

NaN and Infinities have different uses. NaN is the lack of a value for a metric, used when it's inherently incalculable. Infinities are generally actual values of the metric. 

For instance, log-loss is Infinity when the model is fully confident (probability == 0 or probability == 1), and incorrect, about a prediction (some [background](https://github.com/dotnet/machinelearning/issues/2708#issuecomment-467106396)). 

AutoML․NET ignores NaN metric values because they are not actual values of the metric, and are often caused by a self-induced data issue in how the AutoML code creates its CV splits.

The retuned metric value from cross-validation, among other uses, is meant to be representative of the metric as if trained/scored on a larger dataset. In the case of log-loss == Infinity, somewhere in the CV test datasets, there is one or more rows giving an Infinity value, which would be part of this larger scoring dataset. To be representative of this larger scoring dataset, the output value of the cross-validation should also be Infinity.

Reply:
I see, thanks for the explanantion